plugins/obs-outputs/librtmp/rtmp.c

```diff
@@ -1059,53 +1034,60 @@ RTMP_Connect1(RTMP *r, RTMPPacket *cp)
-    h = gethostbyname("localhost");
-    if (!h && GetLastError() == WSAHOST_NOT_FOUND)
-    {
-        r->last_error_code = WSAHOST_NOT_FOUND;
-        RTMP_Log(RTMP_LOGERROR, "RTMP_Connect: Connection test failed. This error is likely caused by Comodo Internet Security running OBS in sandbox mode. Please add OBS to the Comodo automatic sandbox exclusion list, restart OBS and try again (11001).");
+    //uint64_t connect_start = os_gettime_ns();
+    struct happy_eyeballs_ctx* happy_ctx = NULL;
+    int result = happy_eyeballs_init(&happy_ctx);
+    bool free_hostname = FALSE;
+    char *hostname = NULL;
+    int port = 0;
+    if (result != 0)
+    {
+        r->last_error_code = happy_ctx ? happy_eyeballs_get_error_code(happy_ctx) : ENOMEM;
+        happy_eyeballs_destroy(happy_ctx);
         return FALSE;
     }
-#endif
-
-    memset(&service, 0, sizeof(service));
-
-    if (r->m_bindIP.addrLen)
-        addrlen_hint = r->m_bindIP.addrLen;
-
     if (r->Link.socksport)
     {
         /* Connect via SOCKS */
-        if (!add_addr_info(&service, &addrlen, &r->Link.sockshost, r->Link.socksport, addrlen_hint, &socket_error))
-        {
-            r->last_error_code = socket_error;
-            return FALSE;
-        }
+        free_hostname = get_hostname(&r->Link.sockshost, &hostname);
+        port = r->Link.socksport;
     }
     else
     {
         /* Connect directly */
-        if (!add_addr_info(&service, &addrlen, &r->Link.hostname, r->Link.port, addrlen_hint, &socket_error))
-        {
-            r->last_error_code = socket_error;
-            return FALSE;
-        }
+        free_hostname = get_hostname(&r->Link.hostname, &hostname);
+        port = r->Link.port;
     }
+    // Set local bind address (if present)
+    happy_eyeballs_set_bind_addr(happy_ctx, r->m_bindIP.addrLen, &r->m_bindIP.addr);
 
-    if (!RTMP_Connect0(r, (struct sockaddr *)&service, addrlen))
+    // Attempt connection
+    result = happy_eyeballs_connect(happy_ctx, hostname, port);
+    if (free_hostname) {
+        free(hostname);
+        hostname = NULL;
+    }
+    if (result != 0 && result != EAGAIN) {
+        // Error in happy_eyeballs_connect
+        r->last_error_code = happy_eyeballs_get_error_code(happy_ctx);
+        happy_eyeballs_destroy(happy_ctx);
+        return FALSE;
+    }
+    result = happy_eyeballs_timedwait_default(happy_ctx);
+    if (result == ETIMEDOUT) {
+        r->last_error_code = ETIMEDOUT;
+        happy_eyeballs_destroy(happy_ctx);
+        return FALSE;
+    } else if (result != 0) {
+        // Error in happy_eyeballs_timedwait_default
+        r->last_error_code = happy_eyeballs_get_error_code(happy_ctx);
+        happy_eyeballs_destroy(happy_ctx);
+        return FALSE;
+    }
+    // Successful connection
+    int socket_fd = happy_eyeballs_get_socket_fd(happy_ctx);
+    happy_eyeballs_destroy(happy_ctx);
+    if (!RTMP_Connect0(r, socket_fd))
```

Comment:
No idea if this is clearer.. I wrote it out and feel a bit ambivalent. Food for thought
```cpp
    result = FALSE;
    he_result = happy_eyeballs_connect(happy_ctx, hostname, port);
    if (he_result != 0 && he_result != EAGAIN) {
        // Error in happy_eyeballs_connect
        r->last_error_code = happy_eyeballs_get_error_code(happy_ctx);
        goto exit;
    }
    result = happy_eyeballs_timedwait_default(happy_ctx);
    if (result == ETIMEDOUT) {
        r->last_error_code = ETIMEDOUT;
        goto exit;
    } else if (result != 0) {
        // Error in happy_eyeballs_timedwait_default
        r->last_error_code = happy_eyeballs_get_error_code(happy_ctx);
        goto exit;
    }
    // Successful connection
    int socket_fd = happy_eyeballs_get_socket_fd(happy_ctx);
    result = RTMP_Connect0(r, socket_fd);
    if (result)
        r->m_bSendCounter = TRUE;
exit:
    if (happy_ctx)
        happy_eyeballs_destroy(happy_ctx);
    if (free_hostname)
        free(free_hostname);
    
    return result;
```